base/random.jl

```diff
@@ -1194,6 +1196,9 @@ The `Base` module currently provides an implementation for the types
     end
 end
 
+Base.@irrational SQRT_HALF 0.7071067811865475244008  sqrt(big(0.5))
+randn{T}(rng::AbstractRNG, ::Type{Complex{T}}) = Complex{T}(SQRT_HALF * randn(rng, T), SQRT_HALF*randn(rng, T))
```

Comment:
Ah right I forgot about the new `where` syntax. Good call about restriction at the call site. I wasn't sure if should just let it blow up at the call the `randn(rng, T)`, but it does makes the error message a little more cryptic doesn't it.